azext_iot/operations/hub.py

```diff
@@ -1345,3 +1345,56 @@ def _iot_hub_monitor_feedback(target, device_id, wait_on_id):
+
+
+def iot_hub_distributed_tracing_show(cmd, hub_name, device_id, resource_group_name=None):
+    device_twin = _iot_hub_distributed_tracing_show(cmd, hub_name, device_id, resource_group_name)
+    return _customize_device_tracing_output(device_twin['deviceId'], device_twin['properties']['desired'],
+                                            device_twin['properties']['reported'])
+
+
+def iot_hub_distributed_tracing_update(cmd, hub_name, device_id, sampling_mode, sampling_rate,
+                                       resource_group_name=None):
+    if int(sampling_rate) not in range(0, 100):
+        raise CLIError('Sampling rate is a percentage, So only values from 0 to 100(inclusive) are permitted.')
+    device_twin = _iot_hub_distributed_tracing_show(cmd, hub_name, device_id, resource_group_name)
+    if TRACING_PROPERTY not in device_twin['properties']['desired']:
+        device_twin['properties']['desired'][TRACING_PROPERTY] = {}
+    device_twin['properties']['desired'][TRACING_PROPERTY]['sampling_rate'] = int(sampling_rate)
+    device_twin['properties']['desired'][TRACING_PROPERTY]['sampling_mode'] = 1 if sampling_mode.lower() == 'on' else 2
+    result = iot_device_twin_update(cmd, device_id, device_twin, hub_name, resource_group_name)
+    return _customize_device_tracing_output(result.device_id, result.properties.desired,
+                                            result.properties.reported)
+
+
+def _iot_hub_distributed_tracing_show(cmd, hub_name, device_id, resource_group_name=None):
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name)
+    device_twin = iot_device_twin_show(cmd, device_id, hub_name, resource_group_name)
+    _validate_device_tracing(target, device_twin)
+    return device_twin
+
+
+def _validate_device_tracing(target, device_twin):
+    if target['location'].lower() not in ['northeurope', 'westus2', 'west us 2', 'southeastasia']:
+        raise CLIError('Tracing doesn\'t supported for the hub located at "{}" location.'.format(target['location']))
+    if target['sku_tier'].lower() != 'standard':
+        raise CLIError('Tracing doesn\'t supported for the hub belongs to "{}" sku tier.'.format(target['sku_tier']))
+    if device_twin['capabilities']['iotEdge']:
+        raise CLIError('The device "{}" should be non-edge device.'.format(device_twin['deviceId']))
+
+
+def _customize_device_tracing_output(device_id, desired, reported):
+    output = {}
+    desired_tracing = desired.get(TRACING_PROPERTY, None)
+    if desired_tracing:
+        output['Device ID'] = device_id
+        output['Sampling Mode'] = 'enabled' if desired_tracing.get('sampling_mode') == 1 else 'disabled'
+        output['Sampling Rate'] = '{}%'.format(desired_tracing.get('sampling_rate'))
+        reported_tracing = reported.get(TRACING_PROPERTY, None)
+        if (reported_tracing and
+                desired_tracing.get('sampling_mode') == reported_tracing.get('sampling_mode').get('value', None) and
```

Comment:
Please remind me if we need this extra indent on the second and third line of the and logic, I don't remember if it is consistent or not

Reply:
it is required to have an indent(space) on next line